lib/components/admin/field-trip-windows.js

```diff
@@ -0,0 +1,245 @@
+
+// List of tabs used for filtering field trips.
+const TABS = [
+  {id: 'new', label: 'New', filter: (req) => req.status !== 'cancelled' && (!req.inboundTripStatus || !req.outboundTripStatus)},
+  {id: 'planned', label: 'Planned', filter: (req) => req.status !== 'cancelled' && req.inboundTripStatus && req.outboundTripStatus},
+  {id: 'cancelled', label: 'Cancelled', filter: (req) => req.status === 'cancelled'},
+  {id: 'past', label: 'Past', filter: (req) => req.travelDate && moment(req.travelDate).diff(moment(), 'days') < 0},
+  {id: 'all', label: 'All', filter: (req) => true}
+]
+
+// List of fields in field trip object to which user's text search input applies.
+const SEARCH_FIELDS = [
+  'address',
+  'ccLastFour',
+  'ccName',
+  'ccType',
+  'checkNumber',
+  'city',
+  'classpassId',
+  'emailAddress',
+  'endLocation',
+  'grade',
+  'phoneNumber',
+  'schoolName',
+  'startLocation',
+  'submitterNotes',
+  'teacherName'
+]
+
+/**
+ * Collects the various draggable windows for the Field Trip module.
+ */
+class FieldTripWindows extends Component {
+  _onClickFieldTrip = (request) => {
+    const {callTaker, fetchFieldTripDetails, setActiveFieldTrip} = this.props
+    if (request.id === callTaker.fieldTrip.activeId) {
+      this._onCloseActiveFieldTrip()
+    } else {
+      setActiveFieldTrip(request.id)
+      fetchFieldTripDetails(request.id)
+    }
+  }
+
+  _onClickRefresh = () => this.props.fetchFieldTrips()
+
+  _onCloseActiveFieldTrip = () => {
+    this.props.setActiveFieldTrip(null)
+  }
+
+  _onSearchChange = e => {
+    this.props.setFieldTripFilter({search: e.target.value})
+  }
+
+  _onTabChange = e => {
+    this.props.setFieldTripFilter({tab: e.currentTarget.name})
+  }
+
+  render () {
+    const {callTaker, toggleFieldTrips} = this.props
+    const {fieldTrip} = callTaker
+    const {activeId, filter} = fieldTrip
+    const {search} = filter
+    const activeTab = TABS.find(tab => tab.id === filter.tab)
+    const visibleRequests = fieldTrip.requests.data
+      .filter(ft => {
+        let isVisible = false
+        // First, filter field trip on whether it should be visible for the
+        // active tab.
+        if (activeTab) isVisible = activeTab.filter(ft)
+        // If search input is found, only include field trips with at least one
+        // field that matches search criteria.
+        if (search) {
+          isVisible = SEARCH_FIELDS.some(key => {
+            const value = ft[key]
+            let hasMatch = false
+            if (value) {
+              hasMatch = value.toLowerCase().indexOf(search.toLowerCase()) !== -1
+            }
+            return hasMatch
+          })
+        }
+        return isVisible
+      })
+    return (
+      <>
+        {fieldTrip.visible
+          ? <DraggableWindow
+            draggableProps={{
+              defaultPosition: fieldTrip.position
+            }}
+            header={
+              <>
+                <h4>
```

Comment:
Would it make sense for this heading to be a (styled) `<h3>`? (see same remark for `FieldTripDetails`.